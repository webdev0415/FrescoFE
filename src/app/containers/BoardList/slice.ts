import { PayloadAction } from '@reduxjs/toolkit';
import { createSlice } from 'utils/@reduxjs/toolkit';
import { ContainerState } from './types';

// The initial state of the BoardList container
export const initialState: ContainerState = {
  loading: false,
  boardList: [],
};

const boardListSlice = createSlice({
  name: 'boardList',
  initialState,
  reducers: {
    attemptGetBoards(state, action) {
      state.loading = true;
    },
    getBoardsSuccess(state, action: PayloadAction<any>) {
      state.boardList = action.payload;
      state.loading = false;
    },
    getBoardsFail(state) {
      state.boardList = [];
      state.loading = false;
    },
    deleteBoard(state, action) {
      console.log('action', action);
      console.log('state.boardList', state.boardList);
      state.boardList = state.boardList.filter(
        board => board.id !== action.payload,
      );
    },
  },
});

export const { actions, reducer, name: sliceKey } = boardListSlice;
