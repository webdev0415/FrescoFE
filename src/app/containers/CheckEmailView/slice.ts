import { PayloadAction } from '@reduxjs/toolkit';
import { createSlice } from 'utils/@reduxjs/toolkit';
import { ContainerState } from './types';

// The initial state of the CheckEmailView container
export const initialState: ContainerState = {};

const checkEmailViewSlice = createSlice({
  name: 'checkEmailView',
  initialState,
  reducers: {
    someAction(state, action: PayloadAction<any>) {},
  },
});

export const { actions, reducer, name: sliceKey } = checkEmailViewSlice;
