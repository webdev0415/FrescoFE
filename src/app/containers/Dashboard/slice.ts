import { PayloadAction } from '@reduxjs/toolkit';
import { createSlice } from 'utils/@reduxjs/toolkit';
import { ContainerState } from './types';

// The initial state of the Dashboard container
export const initialState: ContainerState = {
  loading: false,
  listEmail: [],
  linkInvitation: '',
};

const dashboardSlice = createSlice({
  name: 'dashboard',
  initialState,
  reducers: {
    searchEmailRequest(state, action: PayloadAction<any>) {
      state.linkInvitation = '';
      // state.loading = true;
    },

    // TODO: define Payload type and use here
    searchEmailSuccess(state, action: PayloadAction<any>) {
      // console.log(action.payload);
      state.listEmail = action.payload;
      // state.loading = false;
    },

    searchEmailError(state, action: PayloadAction<any>) {
      // state.loading = false;
    },

    invitationRequest(state, action: PayloadAction<any>) {
      state.linkInvitation = '';
      state.loading = true;
    },

    // TODO: define Payload type and use here
    invitationSuccess(state, action: PayloadAction<any>) {
      state.linkInvitation = action.payload;
      state.loading = false;
    },

    invitationError(state, action: PayloadAction<any>) {
      state.loading = false;
    },
  },
});

export const { actions, reducer, name: sliceKey } = dashboardSlice;
