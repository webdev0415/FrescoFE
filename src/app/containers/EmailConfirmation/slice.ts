import { PayloadAction } from '@reduxjs/toolkit';
import { createSlice } from 'utils/@reduxjs/toolkit';
import { ContainerState } from './types';

// The initial state of the EmailConfirmation container
export const initialState: ContainerState = {
  loading: false,
};

const emailConfirmationSlice = createSlice({
  name: 'emailConfirmation',
  initialState,
  reducers: {
    emailConfirmationRequest(state, action: PayloadAction<any>) {
      state.loading = true;
    },
    // TODO: define Payload type and use here
    emailConfirmationSuccess(state, action: PayloadAction<any>) {
      state.loading = false;
    },

    emailConfirmationError(state) {
      state.loading = false;
    },
  },
});

export const { actions, reducer, name: sliceKey } = emailConfirmationSlice;
