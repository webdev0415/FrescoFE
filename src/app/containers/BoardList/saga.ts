import { all, put, takeEvery } from 'redux-saga/effects';
import { message } from 'antd';
import { actions } from './slice';
import axios from 'axios';
import Auth from 'services/Auth';
axios.defaults.headers.common['Authorization'] = `Bearer ${Auth.getToken()}`;

// import { actions as globalActions } from '../../slice';
export function* getBoards(action) {
  const { payload } = action;

  try {
    axios.defaults.headers.common[
      'Authorization'
    ] = `Bearer ${Auth.getToken()}`;
    const response = yield axios.get(`/board/organization/${payload}`);

    yield put(actions.getBoardsSuccess(response.data));
  } catch (error) {
    yield put(actions.getBoardsFail());
    message.error(error.message);
  }
}

export function* boardListSaga() {
  yield all([takeEvery(actions.attemptGetBoards.type, getBoards)]);
}
