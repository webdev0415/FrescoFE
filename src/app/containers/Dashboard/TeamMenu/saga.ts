import { all, put, takeLatest } from 'redux-saga/effects';
import { message } from 'antd';
import axios from 'axios';
import { actions } from './slice';

export function* teamMenu(action) {
  const { payload } = action;
  const { orgId, token } = payload;
  try {
    axios.defaults.headers.common['Authorization'] = `Bearer ${token}`;
    const response = yield axios.get('teams?orgId=' + orgId.orgId);
    yield put(actions.getTeamMenuSuccess(response.data));
  } catch (error) {
    yield put(
      actions.getTeamMenuError({
        message: error.message,
        status: error.response.status,
      }),
    );
    message.error(error.message);
  }
}

export function* teamMenuSaga() {
  yield all([takeLatest(actions.getTeamMenuRequest.type, teamMenu)]);
}
