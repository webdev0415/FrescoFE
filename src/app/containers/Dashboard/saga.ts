import { all, put, takeLatest } from 'redux-saga/effects';
import { message } from 'antd';
import axios from 'axios';
import { actions } from './slice';
import { v4 as uuidv4 } from 'uuid';

// import { actions as globalActions } from '../../slice';

export function* searchEmail(action) {
  const { payload } = action;
  // console.log('payload', payload);
  try {
    axios.defaults.headers.common['Authorization'] = `Bearer ${payload.token}`;
    const response = yield axios.post('/users/suggest', {
      email: payload.data.email,
      orgId: payload.data.orgId,
    });
    // console.log('response', response);
    yield put(actions.searchEmailSuccess(response.data));
  } catch (error) {
    yield put(
      actions.searchEmailError({
        message: error.message,
        status: error.response.status,
      }),
    );
    message.error(error.message);
  }
}

export function* invitation(action) {
  const { payload } = action;
  const { history, permission, token, orgId, data } = payload;
  const feToken = uuidv4();
  // console.log('payload', data, payload?.data[0], feToken);
  try {
    axios.defaults.headers.common['Authorization'] = `Bearer ${token}`;
    const response = yield axios.post('invitation', {
      toUserId: data[0]?.id || '',
      toEmail: data[0]?.email || data.email,
      token: feToken,
      permission,
      orgId,
    });
    // console.log('response', response);
    yield put(actions.invitationSuccess(response.data));
    message.success('Invitation successfully.');
  } catch (error) {
    history.push('/err');
    yield put(
      actions.invitationError({
        message: error.message,
        status: error.response.status,
      }),
    );
    message.error(error.message);
  }
}

export function* dashboardSaga() {
  yield all([
    takeLatest(actions.searchEmailRequest.type, searchEmail),
    takeLatest(actions.invitationRequest.type, invitation),
  ]);
}
