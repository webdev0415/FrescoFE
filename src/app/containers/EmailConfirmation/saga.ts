import { takeLatest } from 'redux-saga/effects';
import { message } from 'antd';
import axios from 'axios';
import { actions } from './slice';

// import { actions as globalActions } from '../../slice';

export function* emailConfirmation(action) {
  const { payload } = action;
  const { history } = payload;
  try {
    yield axios.get(`auth/confirm/${payload.code}`);
    message.success('Email is confirmed. Please login.');
    history.push('/auth/login');
  } catch (error) {
    history.push('/auth/login');
    message.error(error.message);
    message.error('Please try again.');
  }
}

export function* emailConfirmationSaga() {
  yield takeLatest(actions.emailConfirmationRequest.type, emailConfirmation);
}
