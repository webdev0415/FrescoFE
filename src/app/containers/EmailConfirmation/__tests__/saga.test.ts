import { put } from 'redux-saga/effects';
import { actions } from '../slice';

import { emailConfirmation, emailConfirmationSaga } from '../saga';

describe('EmailConfirmation-> Saga-> EmailConfirmation', () => {
  let emailConfirmationSagaIterator: ReturnType<typeof emailConfirmation>;
  beforeEach(() => {
    const data = {
      username: 'example',
      workEmail: 'example@fresco.com',
      password: 'admin123',
    };

    const action = actions.emailConfirmationRequest({
      history: { push: () => {} },
      data,
    });

    emailConfirmationSagaIterator = emailConfirmation(action);

    const { payload } = action;

    const requestDescriptor = emailConfirmationSagaIterator.next(payload).value;

    expect(requestDescriptor).toMatchSnapshot();
  });

  it('emailConfirmation success', () => {
    const iteration = emailConfirmationSagaIterator.next();
    expect(iteration.done).toBe(true);
  });

  it('emailConfirmation error', () => {
    const data = {
      username: 'example',
      workEmail: 'example@fresco.com',
      password: 'admin123',
    };

    const action = actions.emailConfirmationRequest({
      history: { push: () => {} },
      data,
    });

    emailConfirmationSagaIterator = emailConfirmation(action);

    const { payload } = action;

    const requestDescriptor = emailConfirmationSagaIterator.next();
    expect(requestDescriptor).toMatchSnapshot();
    // eslint-disable-next-line @typescript-eslint/no-unused-expressions
    emailConfirmationSagaIterator.throw({
      error: {},
      history: {},
    }).value;

    const iteration = emailConfirmationSagaIterator.next();
    expect(iteration.done).toBe(true);
  });
});
describe('EmailConfirmation-> Saga-> EmailConfirmation', () => {
  it('EmailConfirmationSaga watch all', () => {
    let emailConfirmationWatchAllIterator: ReturnType<typeof emailConfirmationSaga>;
    emailConfirmationWatchAllIterator = emailConfirmationSaga();
    const watchDescriptor = emailConfirmationWatchAllIterator.next().value;
    expect(watchDescriptor).toMatchSnapshot();
  });
});
