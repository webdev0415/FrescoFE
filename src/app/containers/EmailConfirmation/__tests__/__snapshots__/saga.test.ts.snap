// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailConfirmation-> Saga-> EmailConfirmation EmailConfirmationSaga watch all 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "emailConfirmation/emailConfirmationRequest",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "FORK",
}
`;

exports[`EmailConfirmation-> Saga-> EmailConfirmation emailConfirmation error 1`] = `Promise {}`;

exports[`EmailConfirmation-> Saga-> EmailConfirmation emailConfirmation error 2`] = `
Object {
  "done": false,
  "value": Promise {},
}
`;

exports[`EmailConfirmation-> Saga-> EmailConfirmation emailConfirmation success 1`] = `Promise {}`;
